apps/cowswap-frontend/src/modules/swap/helpers/getAmountsForSignature.test.ts

```diff
@@ -67,17 +81,23 @@ describe('getAmountsForSignature()', () => {
       // Just subtracted fee from inputAmount, 3012 - 8 = 3004
       expect(result.inputAmount.toFixed()).toEqual('3004.000000')
 
-      // Fee in WETH = 0.005312084993359893
-      // outputAmount = (2 - 0.005312) - 5% = 1.89495
-      expect(result.outputAmount.toFixed()).toEqual('1.894953519256308100')
+      if (partnerFeeBps === 0) {
+        // outputAmount = 2 - 5% = 1.900000000000000000
+        expect(result.outputAmount.toFixed()).toEqual('1.900000000000000000')
+      } else {
+        // outputAmount = 2 - 5% - 20% = 1.520000000000000000
```

Comment:
this is a bit hard to follow, as you are assuming you get a 20% because there's only two tests and one is zero, and the other 20%.

One other way, could be to add this in the `describe.each`

what about something like:

```js
describe.each([
   {partnerFeeBps: 0, expectedOutputAmount: '1.900000000000000000' }, //  explainer here...
   {partnerFeeBps: 2000, expectedOutputAmount: '1.520000000000000000' }, //  explainer here...
])
```

Reply:
In the above example, fee is constant. I think is good, as these test would focus on testing variations on the partnerFee for sell orders, but we will need some that focus on variations on the fee too.

Alternativelly, you might want to do a mega-test with all your cases:

describe.each([
   {partnerFeeBps: 0,  fee: 0, slippage: 0, expectedOutputAmount: '1.900000000000000000' }, //  explainer here...
   ... kee some params constants and change the others, to assert the result given some combinations
])

Reply:
Refactored the tests, split them into separate cases in WHEN/THEN style